Gateway/Request/Recipient/AbstractRecipientDataBuilder.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Buckaroo\Magento2\Gateway\Request\Recipient;
+
+use Buckaroo\Magento2\Gateway\Request\AbstractDataBuilder;
+use Magento\Sales\Api\Data\OrderAddressInterface;
+
+class AbstractRecipientDataBuilder extends AbstractDataBuilder
+{
+    private string $addressType;
+
+    public function __construct(string $addressType = 'billing')
+    {
+        $this->addressType = $addressType;
+    }
+
+    public function build(array $buildSubject): array
+    {
+        parent::initialize($buildSubject);
+
+        return ['recipient' => $this->buildData()];
+    }
+
+    protected function buildData(): array
+    {
+        return
+            [
+                'category' => $this->getCategory(),
+                'gender' => $this->getGender(),
+                'firstName' => $this->getFirstname(),
+                'lastName' => $this->getLastName(),
+                'birthDate' => $this->getBirthDate()
+            ];
+    }
+
+    protected function getAddress(): OrderAddressInterface
+    {
+        if ($this->addressType == 'shipping') {
+            return $this->getOrder()->getShippingAddress();
+        } else {
+            return $this->getOrder()->getBillingAddress();
+        }
+    }
+
+    protected function getFirstname(): string
+    {
+        return $this->getAddress()->getFirstname();
+    }
+
+    protected function getLastName(): string
+    {
+        return $this->getAddress()->getLastName();
+    }
+
+    protected function getCategory(): string
+    {
+        return 'B2C';
+    }
+
+    protected function getGender(): string
```

Comment:
Maybe we should add extra gender values here according to BP-1933.

Reply:
I change these files:
app/code/Buckaroo/Magento2/Gateway/Request/PayPerEmailDataBuilder.php:37
app/code/Buckaroo/Magento2/Gateway/Request/Recipient/BillinkDataBuilder.php:56